autometa/taxonomy/ncbi.py

```diff
@@ -94,77 +95,81 @@ def __str__(self):
-            Name of provided `taxid` if taxid is found in names.dmp else will return None
+        str
+            Name of provided `taxid` if `taxid` is found in names.dmp else 'unclassified'
 
+        Raises
+        ------
+        ValueError
+            Provided `taxid` is not a positive integer
         """
-        if not self.is_valid_taxid(taxid):
-            return False
+        taxid = self.is_valid_taxid(taxid)
         if not rank:
-            return self.names.get(taxid, 'unclassified')
+            return self.names.get(taxid, "unclassified")
         if rank not in set(NCBI.CANONICAL_RANKS):
-            logger.warning(f'{rank} not in canonical ranks!')
-            return None
+            logger.warning(f"{rank} not in canonical ranks!")
+            return "unclassified"
         ancestor_taxid = taxid
-        if ancestor_taxid == 1:
-            logger.error(
-                f'Taxid cannot be 1! Taxid 1 is the root node which links to itself')
-            return ancestor_taxid
         while ancestor_taxid != 1:
             ancestor_rank = self.rank(ancestor_taxid)
             if ancestor_rank == rank:
-                return self.names.get(ancestor_taxid)
+                return self.names.get(ancestor_taxid, "unclassified")
             ancestor_taxid = self.parent(ancestor_taxid)
+        return self.names.get(ancestor_taxid, "unclassified")
 
     def lineage(self, taxid, canonical=True):
         """
-        Returns the lineage of taxids encountered when traversing to root
+        Returns the lineage of `taxids` encountered when traversing to root
 
         Parameters
         ----------
         taxid : int
-            identifer for a taxon in the Taxonomy Database by NCBI
+            `taxid` in nodes.dmp, whose lineage is being returned
         canonical : bool, optional
-            Returns the names of all the canonical ranks when True and an empty list when False, by default True
+            Lineage includes both canonical and non-canonical ranks when False, and only the canonical ranks when True 
 
         Returns
         -------
         ordered list of dicts
-            [{'taxid':taxid, 'rank':rank,'name':'name'}, ...]
+            [{'taxid':taxid, 'rank':rank,'name':name}, ...]
         """
-        if not self.is_valid_taxid(taxid):
-            return False
         lineage = []
-        while taxid != 1:  # This prevents the 'root' to be added in the list
-            lineage.append({
-                'taxid': taxid,
-                'name': self.name(taxid),
-                'rank': self.rank(taxid)})
+        while taxid != 1:
+            if canonical and self.rank(taxid) not in NCBI.CANONICAL_RANKS:
+                taxid = self.parent(taxid)
+                continue
+            lineage.append(
+                {
+                    "taxid": taxid,
+                    "name": self.name(taxid),
+                    "rank": self.rank(taxid),
+                }
+            )
```

Comment:
Since this function is making use of `rank()` and `parent()` functions which are using `is_valid_taxid()` I thought it would be a bit redundant to run that here again. 
Let me know what you think about it